pkg/network/usm/usm_http2_monitor_test.go

```diff
@@ -428,157 +509,82 @@ func removeHeaderFieldByKey(headerFields []hpack.HeaderField, keyToRemove string
-	if len(setDynamicTableSize) > 0 && setDynamicTableSize[0] {
-		changeDynamicTableSize = true
-
+func (f *framer) writeMultiMessage(t *testing.T, count int, cb func(t *testing.T) *framer) *framer {
+	for i := 0; i < count; i++ {
+		cb(t)
 	}
-	headersFrame, err := usmhttp2.NewHeadersFrameMessage(headerFields, changeDynamicTableSize)
-	require.NoError(t, err, "could not create headers frame")
-
-	for i := 0; i < headerFramesCount; i++ {
-		streamID := 2*i + 1
+	return f
+}
 
-		// Writing the header frames to the buffer using the Framer.
-		require.NoError(t, framer.WriteHeaders(http2.HeadersFrameParam{
-			StreamID:      uint32(streamID),
-			BlockFragment: headersFrame,
-			EndStream:     false,
-			EndHeaders:    true,
-		}), "could not write header frames")
+func (f *framer) Bytes() []byte {
+	return f.buf.Bytes()
+}
 
-		// Writing the data frame to the buffer using the Framer.
-		require.NoError(t, framer.WriteData(uint32(streamID), true, []byte{}), "could not write data frame")
-	}
+func (f *framer) writeSettings(t *testing.T) *framer {
+	require.NoError(t, f.framer.WriteSettings(http2.Setting{}), "could not write settings frame")
+	return f
+}
 
-	return buf.Bytes()
+func (f *framer) writePing(t *testing.T) *framer {
+	require.NoError(t, f.framer.WritePing(true, [8]byte{}), "could not write ping frame")
+	return f
 }
 
-// createMessageWithCustomSettingsFrames creates a message with the given number of settings frames.
-func createMessageWithCustomSettingsFrames(t *testing.T, headerFields []hpack.HeaderField, settingsFramesCount int) []byte {
-	var buf bytes.Buffer
-	framer := http2.NewFramer(&buf, nil)
+func (f *framer) writeRSTStream(t *testing.T, streamID uint32, errCode http2.ErrCode) *framer {
+	require.NoError(t, f.framer.WriteRSTStream(streamID, errCode), "could not write RST_STREAM")
+	return f
+}
 
-	for i := 0; i < settingsFramesCount; i++ {
-		require.NoError(t, framer.WriteSettings(http2.Setting{}), "could not write settings frame")
-	}
+func (f *framer) writeData(t *testing.T, streamID uint32, endStream bool, buf []byte) *framer {
+	require.NoError(t, f.framer.WriteData(streamID, endStream, buf), "could not write data frame")
+	return f
+}
 
-	headersFrame, err := usmhttp2.NewHeadersFrameMessage(headerFields)
-	require.NoError(t, err, "could not create headers frame")
+func (f *framer) writeWindowUpdate(t *testing.T, streamID uint32, increment uint32) *framer {
+	require.NoError(t, f.framer.WriteWindowUpdate(streamID, increment), "could not write window update frame")
+	return f
+}
 
-	// Writing the header frames to the buffer using the Framer.
-	require.NoError(t, framer.WriteHeaders(http2.HeadersFrameParam{
-		StreamID:      uint32(1),
-		BlockFragment: headersFrame,
+func (f *framer) writeRawHeaders(t *testing.T, streamID uint32, headerFrames []byte) *framer {
+	require.NoError(t, f.framer.WriteHeaders(http2.HeadersFrameParam{
+		StreamID:      streamID,
+		BlockFragment: headerFrames,
 		EndStream:     false,
```

Comment:
`EndStream: false,` is redundant (it is the default value)

Reply:
removed